packages/babel-plugin-transform-es2015-modules-commonjs/test/fixtures/strict/export-all/expected.js

```diff
@@ -3,15 +3,51 @@
 Object.defineProperty(exports, "__esModule", {
   value: true
 });
+var _exportNames = {
+  z: true,
+  a: true,
+  b: true,
+  d: true,
+  e: true,
+  f: true,
+  c: true
+};
```

Comment:
This doesn't minify very well compared to using an array 🤔 

I'd prefer a Set, but that'd require a polyfill for all users (not sure if acceptable). `z:!0,` vs `'z',`, `!~_exportNames.indexOf(key)` vs `Object.prototype.hasOwnProperty.call(_exportNames,key)`.

Reply:
Another thing that could be done, as `exports` now always has all exports initialized to `void 0` before this runs, is to check for key presence in `exports`; but that would incorrectly avoid throwing if there are name conflicts from two different `*` reexports.

Reply:
The original PR used an array, but I was concerned with every `export * from` having to iterate `O(NUM_EXPORTS) * O(NUM_REEXPORTS)` times where using the object allows `O(NUM_REEXPORTS)`.

For checking `exports` directly, I'm curious for your thoughts. As far as I know, this is allowed in ES6:

```
// a.js
export const a = "some value";

// b.js
export {a} from "./a";

// c.js
export {a} from "./a";

// d.js
export * from "./b";
export * from "./c";
```
where the final `export * from`s don't conflict because they both re-export the same source module. I was concerned that checking `exports.hasOwnProperty()` would break in those cases. That said, it does certainly complicate things, so if you think that's a case that is worth ignoring, especially since it's pretty edge-casey, I'm happy to do that instead.

Reply:
Right, the duplicates should be allowed if they are `SameValue`, but I think we can't check that at runtime as it'd break with circular references 😞

Reply:
Actually is this a minification issue? Seems like a minifier could centralize the `true` to a variable, e.g.

```
var _exportNames = ['a','b','c'];

var t=true;
var _exportNames = {a:t,b:t,c:t};
```
at which point it's just an extra 11 characters for the declaration.

Reply:
I can always generate it that way too if you want, it's just harder to read.

Reply:
I'm just probably overthinking this, it _is_ (likely) rare corner case, and people bundling for browsers will be using webpack's or rollup's modules anyway.

Reply:
I'd vote to leave it as-is, and we can always optimize it later if we get complaints.